### ISSUE TITLE
docker-machine on macOS 10.12.4 (Mac Pro 3,1 Unsupported)

### ISSUE THREAD

**Issue by czenzel** [Parallels/docker-machine-parallels#62]
Hi, I know I am using an unsupported MacPro 3,1 but I am only having issues creating a docker machine using Parallels with this machine, but all other functions of Parallels work. Can somebody check into this?

```
Christophers-Mac-Pro:~ czenzel$ docker-machine create -d parallels prl-dev
Creating CA: /Users/czenzel/.docker/machine/certs/ca.pem
Creating client certificate: /Users/czenzel/.docker/machine/certs/cert.pem
failed MSpanList_Insert 0xc1a8a0 0x2373e0359f1 0x0
fatal error: MSpanList_Insert

runtime stack:
runtime.MSpanList_Insert(0xb36210, 0xc1a8a0)
	/usr/src/go/src/runtime/mheap.c:692 +0x8f
runtime.MHeap_Alloc(0xb33280, 0x1, 0x1000000000a, 0xdfd9)
	/usr/src/go/src/runtime/mheap.c:240 +0x66
runtime.MCentral_CacheSpan(0xb3a6f8, 0xc1a630)
	/usr/src/go/src/runtime/mcentral.c:85 +0x167
runtime.MCache_Refill(0xc14000, 0xa, 0xc1a630)
	/usr/src/go/src/runtime/mcache.c:90 +0xa0

goroutine 1 [running]:
runtime.switchtoM()
	/usr/src/go/src/runtime/asm_amd64.s:198 fp=0xc2080f0380 sp=0xc2080f0378
runtime.mallocgc(0x90, 0x587ea0, 0xc200000001, 0xc2080ebef0)
	/usr/src/go/src/runtime/malloc.go:178 +0x849 fp=0xc2080f0430 sp=0xc2080f0380
runtime.newarray(0x587ea0, 0x11, 0xc2080ebef0)
	/usr/src/go/src/runtime/malloc.go:365 +0xc1 fp=0xc2080f0468 sp=0xc2080f0430
runtime.makeslice(0x7808e0, 0x11, 0x11, 0x0, 0x0, 0x0)
	/usr/src/go/src/runtime/slice.go:32 +0x15c fp=0xc2080f04b0 sp=0xc2080f0468
math/big.nat.divLarge(0xc20800cdc0, 0x10, 0x25, 0xc2080a2d80, 0x20, 0x24, 0xc2080a2c60, 0x20, 0x24, 0xc208042c80, ...)
	/usr/src/go/src/math/big/nat.go:541 +0x2d0 fp=0xc2080f0648 sp=0xc2080f04b0
math/big.nat.div(0xc20800cdc0, 0x10, 0x25, 0xc2080a2d80, 0x20, 0x24, 0xc2080a2c60, 0x20, 0x24, 0xc208042c80, ...)
	/usr/src/go/src/math/big/nat.go:519 +0x35a fp=0xc2080f06f0 sp=0xc2080f0648
math/big.nat.expNNWindowed(0xc2080a2c60, 0x20, 0x24, 0xc2080437c0, 0x10, 0x14, 0xc208042aa0, 0x10, 0x14, 0xc208042c80, ...)
	/usr/src/go/src/math/big/nat.go:1361 +0xdd7 fp=0xc2080f0980 sp=0xc2080f06f0
math/big.nat.expNN(0xc208043900, 0x10, 0x14, 0xc2080437c0, 0x10, 0x14, 0xc208042aa0, 0x10, 0x14, 0xc208042c80, ...)
	/usr/src/go/src/math/big/nat.go:1260 +0x4b0 fp=0xc2080f0b18 sp=0xc2080f0980
math/big.nat.probablyPrime(0xc208042c80, 0x10, 0x14, 0x14, 0x14)
	/usr/src/go/src/math/big/nat.go:1445 +0x887 fp=0xc2080f0c90 sp=0xc2080f0b18
math/big.(*Int).ProbablyPrime(0xc2080e4b20, 0x14, 0xc2080e4b40)
	/usr/src/go/src/math/big/int.go:741 +0x4a fp=0xc2080f0cc0 sp=0xc2080f0c90
crypto/rand.Prime(0xc25d48, 0xc20803c3c0, 0x400, 0xc2080e4b20, 0x0, 0x0)
	/usr/src/go/src/crypto/rand/util.go:99 +0x479 fp=0xc2080f0d70 sp=0xc2080f0cc0
crypto/rsa.GenerateMultiPrimeKey(0xc25d48, 0xc20803c3c0, 0x2, 0x800, 0xc2080e36e0, 0x0, 0x0)
	/usr/src/go/src/crypto/rsa/rsa.go:187 +0x269 fp=0xc2080f0e70 sp=0xc2080f0d70
crypto/rsa.GenerateKey(0xc25d48, 0xc20803c3c0, 0x800, 0x2f, 0x0, 0x0)
	/usr/src/go/src/crypto/rsa/rsa.go:145 +0x57 fp=0xc2080f0eb0 sp=0xc2080f0e70
github.com/docker/machine/utils.GenerateCert(0xc2080f1250, 0x1, 0x1, 0xc2080b0f90, 0x2d, 0xc2080b1050, 0x2c, 0xc2080b0e10, 0x2b, 0xc2080b0ed0, ...)
	/go/src/github.com/docker/machine/utils/certs.go:137 +0x20e fp=0xc2080f1108 sp=0xc2080f0eb0
github.com/docker/machine/commands.setupCertificates(0xc2080b0e10, 0x2b, 0xc2080b0ed0, 0x2f, 0xc2080b0f90, 0x2d, 0xc2080b1050, 0x2c, 0x0, 0x0)
	/go/src/github.com/docker/machine/commands/commands.go:153 +0x96a fp=0xc2080f12c8 sp=0xc2080f1108
github.com/docker/machine/commands.cmdCreate(0xc2080880c0)
	/go/src/github.com/docker/machine/commands/create.go:49 +0x438 fp=0xc2080f16f8 sp=0xc2080f12c8
github.com/codegangsta/cli.Command.Run(0x799870, 0x6, 0x0, 0x0, 0x7e4bd0, 0x10, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/docker/machine/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:113 +0x1038 fp=0xc2080f1ab8 sp=0xc2080f16f8
github.com/codegangsta/cli.(*App).Run(0xc20803e9c0, 0xc20800a000, 0x5, 0x5, 0x0, 0x0)
	/go/src/github.com/docker/machine/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:156 +0xcf7 fp=0xc2080f1d30 sp=0xc2080f1ab8
main.main()
	/go/src/github.com/docker/machine/main.go:114 +0x9bc fp=0xc2080f1f98 sp=0xc2080f1d30
runtime.main()
	/usr/src/go/src/runtime/proc.go:63 +0xf3 fp=0xc2080f1fe0 sp=0xc2080f1f98
runtime.goexit()
	/usr/src/go/src/runtime/asm_amd64.s:2232 +0x1 fp=0xc2080f1fe8 sp=0xc2080f1fe0

goroutine 5 [syscall]:
os/signal.loop()
	/usr/src/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/src/go/src/os/signal/signal_unix.go:27 +0x35
```

**Comment by romankulikov** [comment]
Is this issue persistent or accidental? It doesn't look like specific to Parallels driver but something related to core of docker-machine or even go runtime.

**Comment by czenzel** [comment]
Hi @romankulikov The issue seems persistent. I had some issues as well on my MacBook Pro (supported 2016 model). It would start the virtual machine but never connect to it from docker-machine. Then when trying over it would error out similarly.

**Comment by legal90** [comment]
Hi, @czenzel. This issue looks similar to https://github.com/docker/docker/issues/28388.
What version of Docker Machine are you using? (run `docker-machine version`)

**Comment by legal90** [comment]
@czenzel Please ensure that you use the latest version of `docker-machine`: https://github.com/docker/machine/releases

Also, try to use the latest `docker-machine-driver-parallels` binary. I've gust released v1.2.3, which was built by go v1.7.5 on macOS 10.12.4.

I'm gonna close close this issue now but let us know if the problem will still exist.
